### DISCUSSION TITLE
main.exe : Behavior of return key changed

### DISCUSSION THREAD

**Discussion by maddes8cht** [ggml-org/llama.cpp#1352] [General]
There is the behavior that the output stops in the middle of the sentence and in interactive mode the input goes back to the user.
Until a few days ago the default behavior of main was that after pressing return the output continued (using Windows).
Even without aborting in the middle of a sentence, i.e. after a correct end of sentence, the active model usually tried to continue the output.
The current behavior is that on the input only an empty line is generated and further waits for an input of the user.
However, it is still not a true multiline input - only empty returns create an empty line.
I consider this behavior to be a combination of the worst possible options.
A real multiline input, as already suggested elsewhere as "author" mode, would be a fine thing.
But the possibility to continue the output with a simple return has also been a nice feature, which has now simply disappeared.

**Comment by DannyDaemonic** [comment]
I think some of the default generation settings changed recently, maybe that's leading to this particular behavior in your model of choice. Try running with `-n -1 --ignore-eos` to see if that helps.
